geo-generic-alg/src/algorithm/line_measures/metric_spaces/euclidean/utils.rs

```diff
@@ -208,22 +438,47 @@ where
     }
 
     if let Some(exterior) = polygon.exterior_ext() {
-        let mut min_dist: F = Float::max_value();
-
-        // Calculate distance to exterior ring using proper line-to-line distance
-        for line1 in linestring.lines() {
-            for line2 in exterior.lines() {
-                let line_dist = distance_line_to_line_generic(&line1, &line2);
-                min_dist = min_dist.min(line_dist);
-
-                if line_dist == F::zero() {
-                    return F::zero();
+        // Check containment logic: if polygon has holes AND first point of LineString is inside exterior ring,
+        // then only consider distance to holes (interior rings). Otherwise, consider distance to exterior.
+        let has_holes = polygon.interiors_ext().next().is_some();
+
+        let first_point_inside = if has_holes {
+            // Check if first point of LineString is inside the exterior ring
+            if let Some(first_coord) = linestring.coords().next() {
+                // Simple point-in-polygon test using ray casting
+                let point_x = first_coord.x();
+                let point_y = first_coord.y();
+                let mut inside = false;
+                let ring_coords: Vec<_> = exterior.coords().collect();
+                let n = ring_coords.len();
+
+                if n > 0 {
+                    let mut j = n - 1;
+                    for i in 0..n {
+                        let xi = ring_coords[i].x();
+                        let yi = ring_coords[i].y();
+                        let xj = ring_coords[j].x();
+                        let yj = ring_coords[j].y();
+
+                        if ((yi > point_y) != (yj > point_y))
+                            && (point_x < (xj - xi) * (point_y - yi) / (yj - yi) + xi)
+                        {
+                            inside = !inside;
+                        }
+                        j = i;
+                    }
                 }
```

Comment:
We directly call `ring_contains_coord` in other places but implement a ray-casting algorithm here from scratch. Would it be better to implement a generic trait-based ring_contains_coord? The main dependency of this function, `coord_pos_relative_to_ring`, is already generic.